### DISCUSSION TITLE
mkdir: cannot create directory ‘/backups/daily/ even running mkdir and chown

### DISCUSSION THREAD

**Discussion by quevon24** [prodrigestivill/docker-postgres-backup-local#95] [Q&A] [answered]
Hi, I followed the readme and tried to run commands to create the directory to store the backups with this:

```
# for default images (debian)
mkdir -p /var/opt/pgbackups && chown -R 999:999 /var/opt/pgbackups
# for alpine images
mkdir -p /var/opt/pgbackups && chown -R 70:70 /var/opt/pgbackups
```

When I run the commands I neet to do it with sudo to make it work, because if don't it throws this error: chown: changing ownership of '/var/opt/pgbackups': Operation not permitted and mkdir: cannot create directory ‘/var/opt/pgbackups’: Permission denied

I don't have postgresql on host, only on container but even running those commands it keeps saying :

2021/07/22 03:39:23 443 cmd: /backup.sh
2021/07/22 03:39:23 443: mkdir: cannot create directory ‘/backups/daily/’
2021/07/22 03:39:23 443: : Permission denied
mkdir: cannot create directory ‘/backups/weekly/’: Permission denied
2021/07/22 03:39:23 443: mkdir: cannot create directory ‘/backups/monthly/’: Permission denied
2021/07/22 03:39:23 443 Exit Status: 1

I'm using a VPS with ubuntu 20.04, Docker version 20.10.7 build f0df350, docker-compose version 1.25.0

I also ran docker-compose exec db /bin/bash and inside it I got uid 70 and gid 70 for postgres user/group but even assignin those uid and gid keeps throwing permission denied. I'm missing a step?

Running ls on /var/opt/pgbackups:

`drwxr-xr-x 2 70 70 4096 Jul 22 14:41 pgbackups`

Running id postgres on docker-compose -f docker-compose.prod.yml exec db /bin/bash:

`uid=70(postgres) gid=70(postgres) groups=70(postgres),70(postgres)`

**Comment by prodrigestivill** [comment, marked as answer]
Hello @quevon24, it looks like a configuration error.

The path `/var/opt/pgbackups` is an example path (it is the one specified in the `docker-compose.yml` example), you will need to specify the local path of the `/backups` volume you configured. Also those commands are meant to be run outside the docker itself and with the correct user as you did (so sudo might be required if you are using a rooted docker daemon configuration).

For what you are exposing, looks like you did not configure the `volumes` correctly, if you want to use the local folder `/var/opt/pgbackups` for backups as you are configuring. Then ensure this lines are set up in you `postgres-backup-local` container (look at the example in README for more details):
```yaml
        volumes:
            - /var/opt/pgbackups:/backups
```

Also as stated in the README, running as postgresql user is optional, so you can disable it for your setup.
But in order to avoid confusion, I will also add an "Optional" comment in to the README docker-compose example.

Please close the issue if manged to fix your deployment.

**Comment by quevon24** [comment]
I don't know why this happens, but I stopped all containers, removed them and then create and start them from scratch and the problem vanished, thanks for answering so fast and great job!

**Comment by Frotty** [comment]
I used the example docker compose and am getting the same issues.

```
mkdir: cannot create directory ‘/backups/daily/’: Permission denied
mkdir: cannot create directory ‘/backups/weekly/’: Permission denied
mkdir: cannot create directory ‘/backups/monthly/’: Permission denied
```

which makes no sense to me, as the `/backups` folder is only inside the container as defined in 

```
     volumes:
            - /var/opt/pgbackups:/backups
```

Any help?

**Reply by Frotty** [reply]
Okay adding the `:z` modifier to the volume made it work.

**Reply by prodrigestivill** [reply]
Nice!, thanks for sharing the solution. ☺️

This looks like SELinux specific configuration issue.
